segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/ContainerEventProcessorTests.java

```diff
@@ -0,0 +1,420 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.containers;
+
+import io.pravega.common.util.BufferView;
+import io.pravega.common.util.ByteArraySegment;
+import io.pravega.common.util.ReusableLatch;
+import io.pravega.segmentstore.server.ContainerEventProcessor;
+import io.pravega.segmentstore.server.DirectSegmentAccess;
+import io.pravega.segmentstore.server.SegmentMock;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.IntentionalException;
+import io.pravega.test.common.ThreadPooledTestSuite;
+import lombok.Cleanup;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class ContainerEventProcessorTests extends ThreadPooledTestSuite {
+
+    private static final int TIMEOUT_SUITE_MILLIS = 100000;
+    private static final Duration TIMEOUT_FUTURE = Duration.ofSeconds(10);
+    private static final Duration ITERATION_DELAY = Duration.ofMillis(100);
+    private static final Duration CONTAINER_OPERATION_TIMEOUT = Duration.ofMillis(20);
+
+    @Rule
+    public Timeout globalTimeout = new Timeout(TIMEOUT_SUITE_MILLIS, TimeUnit.MILLISECONDS);
+
+    @Override
+    protected int getThreadPoolSize() {
+        return 2;
+    }
+
+    /**
+     * Check that the max number of elements processed per EventProcessor iteration is respected.
+     *
+     * @throws Exception
+     */
+    @Test(timeout = 100000)
```

Comment:
FYI you can reference the `TIMEOUT_SUITE_MILLIS` constant here .

Reply:
Done.